File: experiments/014-LiveTypingFiveRuns/cells/6-LiveTypingRefactoring_2-EvaluateAsLastResource+5-LiveTyping+6-LiveTypingRefactoring+8-DesignHeuristicsInline_free/2019-2c-parcial2-masked/20260917-024002-13645/exercise/java/src/main/java/surveycrawler/CrawlerFacingUp.java
package surveycrawler;

public class CrawlerFacingUp extends CrawlerFacing {

    // facing name

    @Override
    public String facingName() {
        return "Up";
    }

    // moving

    @Override
    public void retreat(SurveyCrawler aCrawler) {
        aCrawler.moveDown();
    }

    @Override
    public void advance(SurveyCrawler aCrawler) {
        aCrawler.moveUp();
    }

    // facing

    @Override
    public void turnCounterClockwise(SurveyCrawler aCrawler) {
        aCrawler.faceLeft();
    }

    @Override
    public void turnClockwise(SurveyCrawler aCrawler) {
        aCrawler.faceRight();
    }
}

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-212555-19812/exercise/python/src/surveycrawler/crawler_facing_up.py
from surveycrawler.crawler_facing import CrawlerFacing


class CrawlerFacingUp(CrawlerFacing):

    # facing name

    @classmethod
    def facing_name(cls):
        return "Up"

    # moving

    def retreat(self, a_crawler):
        a_crawler.move_down()

    def advance(self, a_crawler):
        a_crawler.move_up()

    # facing

    def turn_counter_clockwise(self, a_crawler):
        a_crawler.face_left()

    def turn_clockwise(self, a_crawler):
        a_crawler.face_right()
